### PR TITLE
Use euclidean distance for identifying outliers for lower dimensional features

### PR DESCRIPTION
**Pull request by ulya-tkch**
- [x] code
- [x] tests
- [x] docstring

### CODE REVIEW

File: tests/test_outlier.py [hunk 1]
```diff
@@ -270,6 +270,16 @@ def test_class_public_func():
         OOD_ood_already_fit.params["confident_thresholds"] == confident_thresholds
     ).all()  # Assert not overwritten
 
+    # Testing fit with correct metrics given feature-space size
+    X_small = np.random.rand(20, 3)
+    OOD_minkowski = OutOfDistribution()
+    OOD_minkowski.fit(features=X_small)
+    assert OOD_minkowski.params["knn"].metric == "minkowski"
```
Comment (elisno) [inline tests/test_outlier.py:277]: We're relying on the fact that `NearestNeighbors` uses `p = 2` (by default).
This test seems more appropriate based on the PR title.

```suggestion
    assert OOD_minkowski.params["knn"].effective_metric_ == "euclidean"
```

Comment (ulya-tkch) [reply tests/test_outlier.py]: Good point, this is clearer

File: tests/test_outlier.py [hunk 2]
```diff
@@ -270,6 +270,16 @@ def test_class_public_func():
...
+    # Testing fit with correct metrics given feature-space size
+    X_small = np.random.rand(20, 3)
+    OOD_euclidean = OutOfDistribution()
+    OOD_euclidean.fit(features=X_small)
+    assert OOD_euclidean.params["knn"].metric == "euclidean"
+    X_large = np.random.rand(20, 4)
+    OOD_cosine = OutOfDistribution()
+    OOD_cosine.fit(features=X_large)
+    assert OOD_cosine.params["knn"].metric == "cosine"
+
```
Comment (jwmueller) [inline tests/test_outlier.py:282]: Can you also assert the scores between 0 and 1 under either metric? 

And also if you add one outlier, that its score is closer to 0 than any of the other datapoints' scores?

File: tests/test_outlier.py [hunk 3]
```diff
@@ -270,6 +270,26 @@ def test_class_public_func():
...
+    X_small = np.random.rand(20, 3)
+    OOD_euclidean = OutOfDistribution()
+    OOD_euclidean.fit(features=X_small)
+    assert OOD_euclidean.params["knn"].metric == "euclidean"
+    X_small_with_ood = np.vstack([X_small, [999999.0] * 3])
+    euclidean_score = OOD_euclidean.score(features=X_small_with_ood)
+    assert (euclidean_score <= 1).all()
+    assert np.argmin(euclidean_score) == (euclidean_score.shape[0] - 1)
+
+    # Add OOD datapoint to X_test
```
Comment (jwmueller) [inline tests/test_outlier.py:283]:
```suggestion
    # Re-run with high dimensional dataset to test cosine distance
```

File: tests/test_outlier.py [hunk 4]
```diff
@@ -270,6 +270,26 @@ def test_class_public_func():
...
     ).all()  # Assert not overwritten
 
+    # Testing fit with correct metrics given feature-space size
+    X_small = np.random.rand(20, 3)
+    OOD_euclidean = OutOfDistribution()
+    OOD_euclidean.fit(features=X_small)
+    assert OOD_euclidean.params["knn"].metric == "euclidean"
+    X_small_with_ood = np.vstack([X_small, [999999.0] * 3])
+    euclidean_score = OOD_euclidean.score(features=X_small_with_ood)
+    assert (euclidean_score <= 1).all()
```
Comment (jwmueller) [inline tests/test_outlier.py:280]:
```suggestion
    assert (euclidean_score <= 1).all()
    assert (euclidean_score >= 0).all()
```

File: tests/test_outlier.py [hunk 5]
```diff
@@ -270,6 +270,26 @@ def test_class_public_func():
...
+    assert np.argmin(euclidean_score) == (euclidean_score.shape[0] - 1)
+
+    # Re-run with high dimensional dataset to test cosine distance
+    X_large = np.hstack([np.zeros((200, 400)), np.random.rand(200, 1)])
+    OOD_cosine = OutOfDistribution()
+    OOD_cosine.fit(features=X_large)
+    assert OOD_cosine.params["knn"].metric == "cosine"
+    X_large_with_ood = np.vstack([X_large, [999999.0] * 401])
+    cosine_score = OOD_cosine.score(features=X_large_with_ood)
+    assert (cosine_score <= 1).all()
```
Comment (jwmueller) [inline tests/test_outlier.py:290]:
```suggestion
    assert (cosine_score <= 1).all()
    assert (cosine_score >= 0).all()
```

**Review by jwmueller — APPROVED** [review]
LGTM! just added a small additional assert in unit tests to also ensure scores are nonnegative

**Pull request merged**